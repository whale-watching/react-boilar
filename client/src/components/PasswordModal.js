import React from 'react';
import Modal from 'react-modal';
import { socketEmit } from '../socketEvents';

class PasswordModal extends React.Component {

  state = {
    error: null
  }

  submitPassword = (e) => {
    e.preventDefault();

    const password = e.target.elements.password.value.trim();

    if (!password) {
      return this.setState({ error: 'You must enter password' });
    };

    socketEmit.joinRoom(this.props.roomName, password, (err) => {
      this.setState({ error: err });

      if (!this.state.error) {
        this.props.onRequestClose();
      }
    });

    e.target.elements.password.value = '';
  }

  render() {
    return (
      <Modal
        className="password-modal"
        isOpen={this.props.isOpen}
        onRequestClose={this.props.onRequestClose}
      >
        <form onSubmit={this.submitPassword}>
          <h3>Join Room</h3>
          <p className="error">{this.state.error}</p>
          <p>Password</p>
          <input type="password" name="password" autoFocus autoComplete="off" />
          <button type="submit" className="button-text">Join</button>
        </form>
      </Modal>
    )
  }
}

export default PasswordModal;