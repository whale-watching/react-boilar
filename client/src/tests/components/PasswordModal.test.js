import React from 'react';
import { shallow } from 'enzyme';
import PasswordModal from '../../components/PasswordModal';
import passwordSubmit from '../data/passwordSubmit';

describe('<PasswordModal', () => {
  it('renders PasswordModal correctly', () => {
    const wrapper = shallow(<PasswordModal isOpen={true} onRequestClose={() => { }} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders error if no password is provided', () => {
    const wrapper = shallow(<PasswordModal isOpen={true} onRequestClose={() => { }} />);
    
    wrapper.find('form').simulate('submit', passwordSubmit);

    expect(wrapper.state('error')).toBe('You must enter password');
  });
});