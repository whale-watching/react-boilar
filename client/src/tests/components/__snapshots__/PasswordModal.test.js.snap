// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordModal renders PasswordModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="password-modal"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Join Room
    </h3>
    <p
      className="error"
    />
    <p>
      Password
    </p>
    <input
      autoComplete="off"
      autoFocus={true}
      name="password"
      type="password"
    />
    <button
      className="button-text"
      type="submit"
    >
      Join
    </button>
  </form>
</Modal>
`;
